src/jit/morph.cpp

```diff
@@ -12461,7 +12461,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                         // We need to keep op1 for the side-effects. Hang it off
                         // a GT_COMMA node
 
-                        tree->ChangeOper(GT_COMMA);
+                        tree->ChangeOper(GT_COMMA, GenTree::PRESERVE_VN);
```

Comment:
I'll think about this some more.  I'm think that we  should just not try to do this folding operation during the CSE re-morph phase.  And I also am wondering why the tree didn't fold this the first time it saw these trees during the initial global Morph phase.